packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -596,179 +596,195 @@ class HotRunner extends ResidentRunner {
     if (reason != null) {
       analyticsParameters[kEventReloadReasonParameterName] = reason;
     }
-    for (FlutterDevice device in flutterDevices) {
-      for (FlutterView view in device.views) {
-        if (view.uiIsolate == null)
-          throw 'Application isolate not found';
-      }
-    }
-
+    bool failedReassemble = false;
+    String reloadMessage;
     // The initial launch is from a script snapshot. When we reload from source
     // on top of a script snapshot, the first reload will be a worst case reload
     // because all of the sources will end up being dirty (library paths will
     // change from host path to a device path). Subsequent reloads will
     // not be affected, so we resume reporting reload times on the second
     // reload.
     bool shouldReportReloadTime = !_runningFromSnapshot;
-    final Stopwatch reloadTimer = Stopwatch()..start();
-
-    if (!_isPaused()) {
-      printTrace('Refreshing active FlutterViews before reloading.');
-      await refreshViews();
-    }
-
-    final Stopwatch devFSTimer = Stopwatch()..start();
-    final UpdateFSReport updatedDevFS = await _updateDevFS();
-    // Record time it took to synchronize to DevFS.
-    _addBenchmarkData('hotReloadDevFSSyncMilliseconds', devFSTimer.elapsed.inMilliseconds);
-    if (!updatedDevFS.success)
-      return OperationResult(1, 'DevFS synchronization failed');
-    String reloadMessage;
-    final Stopwatch vmReloadTimer = Stopwatch()..start();
+    final Stopwatch reloadTimer = Stopwatch();
```

Comment:
This is a big diff. Given that our current testing story is inadequate, how confident do you feel about this change?

Reply:
If you view it without whitespace it's a bit smaller (https://github.com/flutter/flutter/pull/32514/files?utf8=✓&diff=split&w=1).  

The risks in this patch are:

- I messed up some timing/benchmarking. That should show up in the benchmarks - this is not expected to cause any regressions on that.
- I messed up some logic and hot reload could result in causing the application to never draw again.  I think the tests I added should help make sure that doesn't happen, but they don't cover any integration scenario of testing this logic in run_hot.dart. However, I think our benchmarks for hot reload would start failing if it really was that broken.

There's definitely potential for some edge case being missed here. If you can think of another test (maybe another integration test) that covers something we don't have that helps this I'd be happy to add it to this PR.